startPanel.js

```diff
@@ -0,0 +1,85 @@
+class StartPanel extends Panel{
+    constructor(application, db)
+    {
+        super(application, db);
+        this.name = "StartPanel"
+        this.db = db;
+    }
+    show(root)
+    {
+      var template = document.querySelector("#StartPage");
+      var templateClone = document.importNode(template.content, true);
+      var configurationButton = templateClone.querySelector("[data-name='ConfigurationPanelButton']");
+      var gameHandler = templateClone.querySelector("[data-name='GameHandler']");
+      var startGameButton = gameHandler.querySelector("[data-name='StartGame']");
+      var stopGameButton = gameHandler.querySelector("[data-name='StopGame']");
+      var teamA = templateClone.querySelector("[data-name='TeamA']");
+      var teamB = templateClone.querySelector("[data-name='TeamB']");
+      root.appendChild(templateClone);
+      this.setTeamsToDropdown(teamA,teamB);
+      var match = new MatchHandler();
+      configurationButton.addEventListener("click", event =>
+    {
+        this.app.sendAction("ConfigurationPanelRequested");
+        clock = null;   
+    })
+    startGameButton.addEventListener("click", event =>
+    {
+        var gameDuration = gameHandler.querySelector("input").value;
+        var SelectedTeams = this.collectTeams(teamA,teamB);
+        if (this.validateConfigration(gameDuration,SelectedTeams) == true)
+        {
+        match.matchStart(gameDuration, SelectedTeams);
+        
+        }
+    })
+    stopGameButton.addEventListener("click", event =>
+    {
+        match.matchStop();
+    })
+    }
+    collectTeams(teamA,teamB)
+    {
+        var teams = {a:teamA.value, b:teamB.value}
+        return teams;
+    }
+    setTeamsToDropdown(teamA,teamB)
+    {
+        var teamList = this.db.getTeams();
+        teamList.forEach(team => 
+            {
+                var optionA = document.createElement("option");
+                var optionB = document.createElement("option");
+                optionA.text = team.name;
+                optionB.text = team.name
+                teamA.add(optionA);
+                teamB.add(optionB);
+            });
+    }
+    validateConfigration(gameDuration,SelectedTeams)
+    {
+        var flag = true;
+        var time = parseInt(gameDuration)
+        if (isNaN(time))
+        {
+            flag = false;
+        }
+        else
+        {
+            flag = true;
+        }
+        switch (true)
```

Comment:
switche ktore nie mają `break`ów to jest zawsze coś pojebanego w kodzie ;) https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#What_happens_if_I_forgot_a_break - czy na pewno Ci o to chodzio?

Reply:
Poprawione